### PR TITLE
Issue4 Required fields are now showing on click create button

### PR DESCRIPTION
**Pull request by navyasrim**
Issue 4 : Mandatory fields should be indicated and a popup should happen that says “Please complete all mandatory fields” if they aren’t. Currently they only do this if a person clicks in a field and doesn’t enter anything. They should show mandatory fields that aren’t completed when a person clicks “Create”

Fixed this issue by removing the OnFocus and implemented the logic in Create button click event. The required fields will also show in toast message.
wk12_userprofiletesting

### CODE REVIEW

**Comment by aaronleechan** [comment]
navyasrim  : 
Please remove the hard coding message from <FormFeedback>{XXXXXX}</FormFeedback>
Please take the error message from this.state.formErrors
You just need to pop up the highlight and error message from the state.

Phone number is also not working properly after user fill up the number.

File: src/components/UserProfile/AddNewUserProfile/UserProfileAdd.jsx [hunk 1]
```diff
@@ -430,14 +436,25 @@ class AddUserProfile extends Component {
     if (googleDoc) {
       userData.adminLinks.push({ Name: 'Google Doc', Link: googleDoc })
     }
-
-    createUser(userData)
+    if(this.state.userProfile.phoneNumber !== null)
```
Comment (getnithesh) [inline src/components/UserProfile/AddNewUserProfile/UserProfileAdd.jsx:439]: can we move all these conditions to a function and use that function here before calling createUser?

File: src/components/UserProfile/AddNewUserProfile/UserProfileAdd.jsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ import 'react-phone-input-2/lib/style.css'
 
 import classnames from 'classnames'
 import TimeZoneDropDown from '../TimeZoneDropDown'
-
+var patt = '';
```
Comment (ireneclaregs) [inline src/components/UserProfile/AddNewUserProfile/UserProfileAdd.jsx:43]: I dont see any reference of this global variable and if you really need, Can you please update it to "let pattern"?

File: src/components/UserProfile/AddNewUserProfile/UserProfileAdd.jsx (hunk 1 above)
Comment (klbeg) [reply src/components/UserProfile/AddNewUserProfile/UserProfileAdd.jsx]: conditions moved to:  inputsHaveLength()

File: src/components/UserProfile/AddNewUserProfile/UserProfileAdd.jsx (hunk 2 above)
Comment (klbeg) [reply src/components/UserProfile/AddNewUserProfile/UserProfileAdd.jsx]: changed 'part' to a const and removed duplicate occurrences

**Comment by klbeg** [comment]
@aaronleechan hard coding of errors is needed if the input fields haven't been focused, but if they have been, formErrors are now displayed instead.

Errors now displayed if users don't focus any fields

'Phone number required" message now appears even if the user adds and deletes a number

Toast error of  "please complete all mandatory fields" now shows if fields are left blank

**Comment by one-community** [comment]
Aaron has wrapped up his work with us. So let me know if this is ready and I'll ask Irene to review it.

**Comment by klbeg** [comment]
Ok.  I'll finalize tomorrow.  I want to double check the code I pushed this morning. 

Thank you!

**Comment by one-community** [comment]
Sounds good, thank you.

**Review by ireneclaregs — APPROVED** [review]
LGTM

**Pull request merged**